src/makeRelativeDate.ts

```diff
@@ -0,0 +1,164 @@
+type RelativeDateOptions = {
+	notAfter?: number;
+	format?: FormatType;
+	showTime?: boolean;
+};
+
+type FormatType = 'short' | 'med' | 'long';
+type PeriodType = 's' | 'm' | 'h' | 'd';
+
+const dayOfWeek = (day: number): string =>
+	[
+		'Sunday',
+		'Monday',
+		'Tuesday',
+		'Wednesday',
+		'Thursday',
+		'Friday',
+		'Saturday',
+	][day];
+
+const monthAbbr = (month: number): string =>
+	[
+		'Jan',
+		'Feb',
+		'Mar',
+		'Apr',
+		'May',
+		'Jun',
+		'Jul',
+		'Aug',
+		'Sep',
+		'Oct',
+		'Nov',
+		'Dec',
+	][month];
+
+const pad = (n: number): number | string => (n < 10 ? `0${n}` : n);
```

Comment:
can this be replaced with native [`padStart`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart)?